configure.py

```diff
@@ -805,7 +805,7 @@ def get_nasm_version(asm):
 
 def get_llvm_version(cc):
   return get_version_helper(
-    cc, r"(^(?:FreeBSD )?clang version|based on LLVM) ([0-9]+\.[0-9]+)")
+    cc, r"(^(?:.+ )?clang version|based on LLVM) ([0-9]+\.[0-9]+)")
```

Comment:
For detecting the support of OpenSSL's functionality, `configure.py` checks `xcode_version>=5.0` or `llvm_version>=3.3` in https://github.com/nodejs/node/blob/master/configure.py#L1330
Thus, it changes the detection result for ex. `Apple clang version 4.0`.

If you need to check for `Alpine`, below one would be better.
```diff
- cc, r"(^(?:FreeBSD )?clang version|based on LLVM) ([0-9]+\.[0-9]+)")
+ cc, r"(^(?:(?:FreeBSD|Alpine) )?clang version|based on LLVM) ([0-9]+\.[0-9]+)")
```

Reply:
That's what this PR is intended to fix. It won't be feasible to hard code the regex string for specific CLANG_VENDORs. Because what if I want to compile clang myself and I set my CLANG_VENDOR to "MyClang". Now I have to recompile Nodejs for whatever distribution I'm using - and I haven't even changed anything specifically related to Nodejs's functionality or behavior. The check here only cares about returning a version string - it doesn't care about the vendor; so there should be no need to do anything other than match against a WORD.

I also compiled nodejs using this branch on my Macbook Pro and it worked just fine. :smile:

Reply:
Sorry for late response.
The problem I pointed out does not occur now, because `Apple LLVM >= 10` is the supported version.